include/sampleflow/producers/differential_evaluation_mh.h

```diff
@@ -0,0 +1,201 @@
+         * samples $x_{i, k}$ that are passed through the signal of the
+         * base class to Consumer objects.
+         *
+         * @param[in] starting_points The initial samples of $x_{i, 0}$ for
+         *   each chain {i}. Must have at least 3 elements.
+         * @param[in] log_likelihood A function object that, when called
+         *   with a sample $x$, returns $\log(\pi(x))$, i.e., the natural
+         *   logarithm of the likelihood function evaluated at the sample.
+         * @param[in] perturb A function object that, when given a sample
+         *   $x$ as well as a vector of rejected samples $y$,
+         *   returns a pair of values containing the following:
+         *   <ol>
+         *   <li> A different sample $\tilde x$ that is perturbed
+         *     in some way from the given sample $x$.
+         *   <li> The relative probability of the transition $x\to\tilde x$
+         *     divided by the probability of the transition $\tilde x\to x$.
+         *     Specifically, if the proposal distribution is given by
+         *     $\pi_\text{proposal}(\tilde x|x)$, then the second element
+         *     of the pair returned by the `perturb` argument is
+         *     $\frac{\pi_\text{proposal}(\tilde x|x)}
+         *           {\pi_\text{proposal}(x|\tilde x)}$.
+         *   </ol>
+         *   If samples are from some continuous space, say
+         *   ${\mathbb R}^n$, then the perturbation function is often
+         *   implemented by choosing $\tilde x$ from a neighborhood
+         *   of $x$. If, for example, $\tilde x$ is chosen with a probability
+         *   that only depends on the distance $\|\tilde x-\x\|$ as is often
+         *   done (e.g., uniformly in a disk of a certain radius around
+         *   $x$, or using a Gaussian probability distribution centered at
+         *   $x$), then the ratio returned as second argument is
+         *   $\frac{\pi_\text{proposal}(\tilde x|x)}
+         *           {\pi_\text{proposal}(x|\tilde x)}=1$.
+         * @param[in] crossover A function that accepts the current sample as well
+         *   as two samples from other chains as arguments and combines them to
+         *   generate a new sample. This is the function that adds the
+         *   "differential evaluation" or "differential evolution" to Metropolis
+         *   Hastings. If the `OutputType` belongs to a vector space, the simple
+         *   solution is
+         *   ```
+         *   return current_sample + gamma * (sample_a - sample_b)
+         *   ```
+         *   where `gamma` is a scaling parameter; $\frac{2.38}{\sqrt{2d}}, d = $ the
+         *   dimension of the vector space, is commonly used.
+         * @param[in] n_chains The number of sample chains to produce in parallel.
+         * @param[in] crossover_gap The number of iterations in between crossover
+         *   iterations.
+         * @param[in] n_samples The number of (new) samples to be produced
```

Comment:
Is this actually true? What happens if I have 4 chains and 7 samples?

Reply:
@bangerth Good question. Should the sampler be cut off after the specified `n_samples` is reached? This would mean that the last chain wouldn't produce as many samples as the others. Or is it okay to produce `n_samples + (n_samples % n_chains)` samples?

Reply:
Just terminate after `n_samples`.